api_server/src/main/java/io/zetch/app/web/LocationController.java

```diff
@@ -57,6 +57,18 @@ LocationDto getOneLocation(@PathVariable String name, JwtAuthenticationToken tok
     return locationService.getOne(name).toDto();
   }
 
+  /**
+   * @param name Location's name
+   * @return A location by name
+   */
+  @GetMapping("/{name}/{type}")
+  @Operation(summary = "Search locations by name and type")
+  @SecurityRequirement(name = "OAuth2")
+  Iterable<LocationDto> searchLocation(
+      @PathVariable String name, @PathVariable String type, JwtAuthenticationToken token) {
+    return locationService.search(name, type).stream().map(LocationEntity::toDto).toList();
+  }
```

Comment:
This seems redundant because we already have `GET /locations/{name}` which essentially searches by an exact name match. We can add a query to LocationRepository that could match on different attributes at once.

I could generate something like this but haven't tested:
```java
  @Query(
      """
      select l from LocationEntity l
      where (upper(l.name) like upper(concat('%', ?1, '%')) or
            upper(l.description) like upper(concat('%', ?2, '%')) or
            upper(l.address) like upper(concat('%', ?3, '%'))) and
            upper(l.type) = upper(?4)
      order by l.name
      """)
  List<LocationEntity> findLocation(String name, String description, String address, Type type);
```

Then our search endpoint could accept query variables like `GET /locations?name=abc&description=abc&address=abc&type=museum`

Reply:
1. Should this "?name=abc&description=abc&address=abc&type=museum" by 1 or 4 path variables?
2. Should we want search by OR rather than by AND? It might be painful to need to type the entire description correctly
3. Should we remove GET /location/{name}